src/apps/search-result/search-result.tsx

```diff
@@ -58,11 +59,18 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
   const { facets: campaignFacets } = useGetFacets(q, filters);
 
   // If q changes (eg. in Storybook context)
-  //  then make sure that we reset the entire result set.
+  // then make sure that we reset the entire result set.
   useEffect(() => {
     setResultItems([]);
   }, [q, pageSize, filters]);
 
+  const { track } = useStatistics();
+  useEffect(() => {
+    track("page", { id: 10, name: "On site search string", trackedData: q });
+    // We actaully just want to track if the query changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [q]);
```

Comment:
The only other dependency here would be the track function. Why not include that? Is it varying?

Reply:
Yes - first it's not there and then it is which would call it twice instead of once.